web/src/flow/stages/prompt/PromptStage.ts

```diff
@@ -193,28 +192,29 @@ ${prompt.initialValue}</textarea
                         <label class="pf-c-check__label" for=${id}>${choice}</label>
                     </div> `;
                 })}`;
-            case PromptTypeEnum.AkLocale:
+            case PromptTypeEnum.AkLocale: {
+                const inDebug = rootInterface()?.config?.capabilities.includes(
+                    CapabilitiesEnum.CanDebug,
+                );
+                const locales = inDebug
+                    ? LOCALES
+                    : LOCALES.filter((locale) => locale.code !== "debug");
+                const options = locales.map(
+                    (locale) => html`<option
+                        value=${locale.code}
+                        ?selected=${locale.code === prompt.initialValue}
+                    >
+                        ${locale.code.toUpperCase()} - ${locale.label()}
+                    </option> `,
+                );
+
                 return html`<select class="pf-c-form-control" name="${prompt.fieldKey}">
                     <option value="" ?selected=${prompt.initialValue === ""}>
-                        ${t`Auto-detect (based on your browser)`}
+                        ${msg("Auto-detect (based on your browser)")}
                     </option>
-                    ${LOCALES.filter((locale) => {
-                        // Only show debug locale if debug mode is enabled
-                        if (locale.code === "debug") {
-                            return rootInterface()?.config?.capabilities.includes(
-                                CapabilitiesEnum.CanDebug,
-                            );
-                        }
-                        return true;
-                    }).map((locale) => {
-                        return html`<option
-                            value=${locale.code}
-                            ?selected=${prompt.initialValue === locale.code}
-                        >
-                            ${locale.code.toUpperCase()} - ${locale.label}
-                        </option>`;
-                    })}
+                    ${options}
```

Comment:
This is a little more surgery than usual because it displays the locales to the customer, so I needed to replace a bit of the logic. I also separated out the business logic (I see I missed one) from the component itself, which is important to me as it leads to independent testing.